### DISCUSSION TITLE
NoData alerts: status vs. notification consistency

### DISCUSSION THREAD

**Discussion by lafrech** [grafana/grafana#44472] [Alerting]
I setup alerts to detect outliers in my data. I configured alert state "No Data" for no data or null values.

When I get no data, the alert rule shows as "Normal" with state "Nodata" in the Alert rules panel. My understanding is that since I set "No Data" rather than "Alerting", the alerts stays in "Normal/Nodata" state rather than "Firing/Alerting".

However, the notification email I get for this group of rules includes the NoData error, labeled as "Firing", with a name of DatasourceNoData rather than the name of the rule.

I understand why people would  like to be notified about no data (e.g. https://github.com/grafana/grafana/discussions/37762). But I don't think the notification email is consistent with the Alert rules panel in this case.

I read the docs again

> NoData: Create a new alert DatasourceNoData with the name and UID of the alert rule, and UID of the datasource that returned no data as labels.
> Ok: Set alert rule state to Normal.

I don't remember that from when I configured the alert. I should probably be using Ok. I guess I picked NoData back then because I wanted "keep current state" rather than "set to Normal" in case of no data. Like, if alert was firing, keep it firing, and the doc has been improved since then.

Anyway, I still find it confusing that the alert appears as Normal in the panel and Firing in the email. And I find the email a bit confusing. Perhaps, rather than "Firing | DatasourceNoData", I'd expect "NoData | Original name of my alert".

I hope I'm being helpful. Thanks for the excellent work on the alerts module.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
